### DISCUSSION TITLE
7.3.0: RS build bug or Webpack file size limit?

### DISCUSSION THREAD

**Discussion by warloque** [react-static/react-static#1448] [General]
After RS has been upgraded to v7.3.0 `build` step works somehow unstable: 
all app's files get built, but it seems like when js file exceeds sort of 'code limit', the app builds but with nested dirs and chunks in `dist`. 
BTW, the same behavior happens for any syntactic error in my code which is quite normal, I guess.

So, for example, an app has two containers (`src/containers/Homepage` and `src/containers/ErrorPage`). Some component imports a helper function from `src/utils/helper.js` - a simple js file written in es6). 
I've got a strong feeling of webpack or RS processing **if...else** and **nested if..else** in a wrong way, because when I remove the nesting or simplify statements' code in those conditions, the app gets built normally.

![Screenshot 2020-06-01 at 19 01 10](https://user-images.githubusercontent.com/375734/83428046-60f5c780-a43a-11ea-8870-a8c68f9426b7.png)

May there exist anything like _webpack file size limit_ or some _code preprocessing/code splitting_ which may result in such strange behavior? 

Thanks in advance!

**Comment by SleeplessByte** [comment]
I don't think any of this changed in the past few releases!

**Reply by warloque** [reply]
investigating my code line-by-line I've discovered that string concatenation fails build...
example of failing string:`Main.setAttribute('style', `margin-top: ${hh}`);`

**Reply by warloque** [reply]
I've also found that deeply nested components when imported also bring those build 'glitches'. 
Especially files containing multiple exports and one default export.

**Reply by SleeplessByte** [reply]
It all sounds like side-effects :/ I don't know how we would figure out what exactly is causing this.
